aws/internal/tfawsresource/sync.go

```diff
@@ -0,0 +1,44 @@
+package tfawsresource
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+)
+
+// Semaphore can be used to limit concurrent executions. This can be used to work with resources with low quotas
+type Semaphore chan struct{}
```

Comment:
For other readers, this will likely be reworked in the future and will receive additional code comments that it is not intended for wide use right now.